README.md

```diff
@@ -44,14 +44,35 @@ left to you. The following are the `Env` methods you should know:
 
 ## Installation
 
+There are two main options for the installation of `scigym`:
+
+#### (a) minimal install (recommended)
+
+This method allows you to install the package with no environment specific dependencies, and later add the dependencies for specific environments as you need them.
+
 You can perform a minimal install of `scigym` with:
 
   ```sh
   git clone https://github.com/hendrikpn/scigym.git
   cd scigym
   pip install -e .
   ```
-You will need additional dependencies for some environments.
+
+To later add the dependencies for a particular `environment_name`, run the following command from the folder containing `setup.py`:
+
+  ```sh
+  pip install -e .[environment_name]
+  ```
+
+#### (b) full install
+
+This method allows you to install the package, along with all dependencies required for all environments. Be careful, scigym is growing, and this method may install a large number of packages. To view all packages that will be installed during a full install, see the `requirements.txt` file in the root directory. If you wish to perform a full installation you can run:
+
+  ```sh
+  git clone https://github.com/hendrikpn/scigym.git
+  cd scigym
+  pip install -e .[all]
```

Comment:
Cool, I didn't know about the [all] method. Does the [all] comment also install the `requirements.txt`? Is this an alternative to `pip install -r requirements.txt`?

Reply:
Yip, this is neat! This functionality is not generic, but only works because of the following lines in `setup.py`:

```
# Meta dependency groups.
all_deps = []
for group_name in extras:
    all_deps += extras[group_name]
extras['all'] = all_deps
```
These lines basically create an additional entry in the `extras` dictionary which contains all the dependencies, which then lets

```
pip install -e .[all]
```
work in the same way as

```
pip install -e .[environment_name]
```

Reply:
Personally, I would not go down that rabbit hole of listing all children dependencies, and I don't think it is common practice. Since you have found this very neat functionality, maybe we can just remove the `requirements.txt` file (as well as its mention in the `README`). Well, unless of course we need it for travis, which I am not sure about.

Reply:
Ok, I think that it would definitely be neatest without the `requirements.txt' file.

I'm not quite sure how to run/test the unit tests with Travis, but if `pip install package_name' works in the `.travis.yml` file then I would guess that replacing all the pip installs with `pip install -e .[all]' should work?

Could you test this, and if it works then we do that, delete the `requirements.txt` file and modify the readme accordingly?

Reply:
Agrred! However, the problem is that `pip install -e .[all]` also installs the package. I am not sure that we want to install the package for testing.

Reply:
Ok, I've now resolved this by creating a structured `test_requirements.txt` file, which is used only by TravisCI for unit testing.

I've updated the readme to indicate that one can look into this file to see a structured list of the dependencies for all environments, and I updated the `CONTRIBUTING.md` guide to explain that for automated unit testing it is necessary to also add environment specific dependencies into `test_requirements.txt`.

I think this resolves everything neatly :)